src/overlays/effects/ovl_Effect_Ss_Blast/z_eff_ss_blast.c

```diff
@@ -7,15 +7,15 @@
 #include "z_eff_ss_blast.h"
 #include "assets/objects/gameplay_keep/gameplay_keep.h"
 
-#define rPrimColorR regs[0]
-#define rPrimColorG regs[1]
-#define rPrimColorB regs[2]
-#define rPrimColorA regs[3]
-#define rEnvColorR regs[4]
-#define rEnvColorG regs[5]
-#define rEnvColorB regs[6]
-#define rEnvColorA regs[7]
-#define rAlphaTarget regs[8]
+#define rInnerColorR regs[0]
+#define rInnerColorG regs[1]
+#define rInnerColorB regs[2]
+#define rInnerColorA regs[3]
+#define rOuterColorR regs[4]
+#define rOuterColorG regs[5]
+#define rOuterColorB regs[6]
+#define rOuterColorA regs[7]
+#define rAlphaStep regs[8]
```

Comment:
Why change only some of these? Isn't the point that these are all used the same way between effects?

Reply:
these regs are effect specific. he only changed the ones where he is sure that prim/env maps to inner/outer. Not all effects will use prim and env for that purpose

Reply:
What does inner/outer mean here?

Reply:
red inner color, blue outer color
![image](https://user-images.githubusercontent.com/22045047/198850240-b34301ed-640a-4528-a9e2-8c258e9013fb.png)

Reply:
But they're still prim and env? I don't see what's inner and outer there, I think that's a much more subjective description than sticking to what they're actually used in.

Reply:
prim and env are just the names of the color channels. The asset will use those channels for different purposes depending on how its mapped. The prim color is in the center of the ring, the inner color. and env is the outside.
Naming it after how the asset uses prim and env is much more helpul information to have

Reply:
If you look at the I8 texture for the shockwave `gEffShockwaveTex`, it's whiteish on the inside of the ring and black on the outside
The combiner set in the dlist `gEffShockwaveDL` lerps from env to prim using that texture as weight (`PRIMITIVE, ENVIRONMENT, TEXEL0, ENVIRONMENT`, `(prim-env)*tex0+env`) so when the texture is white, prim comes out, that is, the color showing on the inside

the inner/outer is "explained" in the main `EffectSs_` function for spawning this effect

Reply:
to take another example from the skulltula actor:
`gDPSetEnvColor(POLY_OPA_DISP++, this->teethR, this->teethG, this->teethB, 0);`
naming this after the fact that its the color of its teeth is much more helpful than saying its the env color. Its about what the prim/env color represent in this context

Reply:
I thought it just LERP'd between them using the texture in the usual way? Much as it does in most of the other effects, I don't see why they all warrant different names.

Reply:
this is going to depend entirely on how the specific effect asset is using the colors. Im not sure what you mean about "the usual way". The colors have context dependent meaning

Reply:
Elliptic are you arguing these regs should be named env/prim ?
If yes, only these regs or also the args in `EffectSs_` functions?
If no, what are we debating

Reply:
I think the point here is that I don't see these as context-dependent because when you look at how they're used, they're pretty much used in the same way as they are in every other effect, as prim and env? As far as I can see, all effect use them for is various different sorts of `gsDPSetCombineLERP` with their textures. If anything, the context is in the combiner mode used rather than what the individual effects look like? I also don't think it will be possible to apply this in general, even if it's apparently the case that here one looks like an "inner" colour and the other an "outer" one?

Reply:
prim and env tell you nothing about how it will visually end up looking/what its purpose is. It does depend on the combiner yes, but the point is to name it in the code after what it will actually end up representing. Pointing back to my previous example, naming that variable as "envColor" tells you nothing, while teeth color does. 

In cases where prim/env clearly map to certain visual aspects of the end result, like an inner and outer color, it makes perfect sense to name it after those things

Reply:
In general I agree with Elliptic that they should just be named prim/env, since how they are used is down to the color combiner and texture. Tho I also agree it can be good to be more explicit in situations where the colors have a very clear purpose. But I don't think this effect in particular falls in the category where it's clearly inner/outer (if anything most of the texture is a mix of both colors afaict), so I would personally keep prim/env here.

Reply:
To bump on this, I think it's nonsense to call things we have names for, "env" and "prim"

There are plenty examples of named colors that show it's much more useful to refer to the use it gets rather than the color register (that's how prim/env/blend/fog colors are called, they're color registers) in use
For example Link's tunic, gauntlets, the actors' colorFilterParams, Volvagia's head/body alpha, Phantom Ganon's eyes brightness, Gohma's eye color, dog colors...

The C equivalent of restricting names to "env" and "prim" would be naming all function arguments after the ABI, for example a0 for every first non-float argument

Reply:
> Can you elaborate on what this would mean for EffectSsBlast_Spawn's args and docs

I would name the args and docs with prim/env, following the names used inside the effect

> To bump on this, I think it's nonsense to call things we have names for, "env" and "prim"

I think it makes sense to use proper names when the purpose of the colors is clear, but I don't really agree that this is the case here. At least looking at the image above I don't see the red and blue being clear inner/outer colors. In fact most of the texture ends up being purple which I assume is because a lot of it is a mix of both prim and env colors. If it was clearly inner/outer, I would expect a part of the ring to be full red, and another part to be full blue, with no mix in between (in terms of IA texture with this combiner, it would mean max alpha on one part, and 0 alpha on the other). At least this is how I interpret what's going on with the texture here.

Reply:
Do you have a better description than inner/outer then?

To be clear, using env/prim is as bad as naming them color1/color2

Reply:
I've been thinking on this for a while now. I'd probably argue that inner/outer while not perfect, are fine as is. The inner color seems to affect the inner portion more (it seems more redish in my eyes), and the same for the outer (at least I see a blueish outline).

Reply:
Of the people who have voiced their opinion, we are 3 to 2 in favor of inner/outer
Id love to come to a better compromise/consensus, but this has been in limbo for a while without much progress. So Id say to go with the majority, even if slight 

ofc anyone open to add their opinion if they havent already